grammars/mlir.json

```diff
@@ -1,112 +1,590 @@
+      "match": "(?<=\\W)\"tfl\\.(cast|ceil|concatenation|densify|dequantize|fill|gather_nd|gather|logistic|pack|quantize|range|reshape|svdf|shape|slice|split|split_v|squeeze|tile|unique|unpack|while|yield)\"(?=\\W)",
+      "name": "entity.name.function.mlir.tflOps"
+    },
+    "tflLongOps": {
+      "match": "(?<=\\W)\"tfl\\.(embedding_lookup|external_const|fake_quant|fully_connected|hard_swish|NumericVerify|one_hot|segment_sum|strided_slice|topk_v2|zeros_like)\"(<=\\W)",
+      "name": "entity.name.function.mlir.tflLongOps"
+    },
+    "tflNormalization": {
+      "match": "(?<=\\W)\"tfl\\.(l2_normalization|local_response_normalization)\"(?=\\W)",
+      "name": "entity.name.function.mlir.tflNormalization"
+    },
+    "tflReduce": {
+      "match": "(?<=\\W)\"tfl\\.(reduce_any|reduce_max|reduce_min|reduce_prod)\"(?=\\W)",
+      "name": "entity.name.function.mlir.tflReduce"
+    },
+    "tflSequence": {
+      "match": "(?<=\\W)\"tfl\\.(reverse_sequence|reverse_v2|unidirectional_sequence_rnn)\"(?=\\W)",
+      "name": "entity.name.function.mlir.tflSequence"
+    },
+    "vectorOps":{
+      "match": "\\Wvector\\.(broadcast|contract|vectorfma|print|constant_mask|create_mask|shuffle|matrix_multiply|outerproduct|reduction|strided_slice|transpose|type_cast)\\b",
+      "name":"entity.name.function.mlir.vectorOps"
+    },
+    "vectorExtract":{
+      "match" : "\\Wvector\\.(extract|extractelement|extract_slices)\\b",
+      "name": "entity.name.function.mlir.vectorExtract"
+    },
+    "vectorInsert":{
+      "match" : "\\Wvector\\.(insert|insertelement|insert_slices|insert_strided_slice)\\b",
+      "name" : "entity.name.function.mlir.vectorInsert"
+    },
+    "vectorReshape":{
+      "match" : "\\Wvector\\.(reshape|shape_cast)\\b",
+      "name": "entity.name.function.mlir.vectorReshape"
+    },
+    "vectorTransfer":{
+      "match" : "\\Wvector\\.transfer_(read|write)\\b",
+      "name" : "entity.name.function.mlir.vectorTransfer"
+    },
+    "vectorTuple":{
+      "match" : "\\Wvector\\.(tuple|tuple_get)\\b",
+      "name" : "entity.name.function.mlir.vectorTuple"
+    },
+    "loopOps":{
+      "match" : "\\bloop\\.(for|if|parallel|reduce|yield)\\b",
+      "name": "entity.name.function.mlir.loopOps"
+    },
+    "tileFunctions":{
+      "match" : "\\btile\\.(constant|contract|index|reshape)\\b",
+      "name": "entity.name.function.mlir.tileFunctions"
+    },
+    "tileQuotedFunctions":{
+      "match" : "(?<=\\W)\"tile\\.(constant|contract|index|reshape)\"(?=\\W)",
+      "name": "entity.name.function.mlir.tileFunctions"
+    },
+    "tileKeywords": {
+      "match": "\\btile\\.(name)\\b",
+      "name": "keyword.other.mlir.tileKeywords"
+    },
+    "eltwiseFunctions":{
+      "match" : "(?<=\\W)\"eltwise\\.(add|div|sconst|cmp_lt|select|sub|exp|cast|neg|mul|sqrt|ident)\"(?=\\W)",
+      "name": "entity.name.function.mlir.tileFunctions"
+    },
+    "CHECK": {
+      "match": "(\/\/)\\s*(CHECK\\s*:|CHECK-\\w+\\s*:)(.*)$",
```

Comment:
I believe "CHECK :" wouldn't trigger FileCheck (please verify) and so should ideally be marked as an error as it is very rarely intentional.